lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -285,6 +285,40 @@ defmodule ExUnit.Callbacks do
   end
 
   @doc """
+  Same as `start_supervised/2` but returns the PID on success and raises if
+  not started porperly.
+  """
+  @spec start_supervised!(Supervisor.child_spec | module | {module, term}, keyword) :: pid
+  def start_supervised!(child_spec_or_module, opts \\ []) do
+    case start_supervised(child_spec_or_module, opts) do
+      {:ok, pid} ->
+        pid
+      {:ok, pid, _info} ->
+        pid
+      {:error, {{:EXIT, {error, stack}}, _info}} ->
```

Comment:
We need to ping @fishcakez here because it is guaranteed we will have an opinion on this and also know what is the best way to proceed.

Reply:
We should not be formatting manually, rather using the `Exception.format_exit` call because we several situations are not handled. It's possible that we don't handle the `{{:EXIT, reason}, info}` in `Exception`, and may need a lighter wrapper that has logic similar to:
```elixir
{:error, {{:EXIT, reason}, info}} ->
  raise """
  #{inspect_info(info)}
  #{Exception.format_exit(reason)}
  """
{:error, reason} ->
  raise """
  #{Exception.format_exit(reason)}
  """
```

Reply:
@fishcakez, sorry for the delay, I'm doing the fix to use `Exception.format_exit/1` now. 

But just a little question: is this `inspect_info/1` function implemented somewhere? The `ExUnit.Callback` does not have it, and since I really have no idea what comes in this tuple, I don't know how to implement it.

Reply:
I think by inspect_info he just meant printing whatever is in info properly. I think it is usually a tuple with three elements, so you can use `Exception.format_mfa` and fallback to inspect if it is not said tuple.

Reply:
Well, looks like this `info` is the child spec, look what we get for an exception on start:

```elixir
{:child, :undefined, ExUnit.SupervisedTest.MyAgent, {ExUnit.SupervisedTest.MyAgent, :start_link, [:exception]}, :permanent, 5000, :worker, [ExUnit.SupervisedTest.MyAgent]}
```

And for a error tuple return on start:

```elixir
{:child, :undefined, ExUnit.SupervisedTest.MyAgent, {ExUnit.SupervisedTest.MyAgent, :start_link, [:error]}, :permanent, 5000, :worker, [ExUnit.SupervisedTest.MyAgent]}
```

Reply:
Any idea how to format it well?

Reply:
I don't think we care about all the other stuff, so I would get the MFA out of it and print it.

It may also be worth checking out what this function will do if we give it an invalid child_spec.

Reply:
OK, I'll implement something like you described and commit the changes!

Reply:
I'll check for invalid child spec then!